op-node/rollup/sequencing/sequencer.go

```diff
@@ -647,8 +647,15 @@ func (d *Sequencer) forceStart() error {
 }
 
 func (d *Sequencer) Stop(ctx context.Context) (hash common.Hash, err error) {
-	if err := d.l.LockCtx(ctx); err != nil {
-		return common.Hash{}, err
+	for {
+		if err := d.l.LockCtx(ctx); err != nil {
+			return common.Hash{}, err
+		}
+		// ensure latestHead has been updated to the latest sealed/gossiped block before stopping the sequencer
+		if d.latestHead.Hash == d.latest.Ref.Hash {
+			break
+		}
```

Comment:
There is no guarantee this happens, as the `latest` has been async-gossiped, but not completed processing. A best attempt is made to process locally, but if e.g. the geth node shuts  down, then this will never complete.
I think we need a handover / conductor-like spec, to ensure we have the right start/stop behavior.

Reply:
It seems much more unlikely that `geth` shuts down or fails to process the FCU event, vs this relatively common race condition. Given `Stop()` is something that is generally used for sequencer handover, IMO we should take the conservative approach here. If geth shuts down mid `Stop()` call, that particular sequencer is borked anyway.